src/io.rs

```diff
@@ -86,6 +91,7 @@ pub fn read_embeddings_view(
         FastText => ReadFastText::read_fasttext(&mut reader).map(Embeddings::into),
         FinalFusion => ReadEmbeddings::read_embeddings(&mut reader),
         FinalFusionMmap => MmapEmbeddings::mmap_embeddings(&mut reader),
+        Floret => bail!("floret embeddings do not provide a view"),
```

Comment:
The backing store should be compatible with views? Is there a reason why we don't want to offer that?

Reply:
IIRC we use views to enable analogy similarity/queries, which doesn't really make sense for floret.

I think we should support views for floret, but then we first need some other typing machinery to make similarity/analogy queries impossible for floret vocabs.

Reply:
I think it'd just yield an empty result set because we're slicing the view up to row `vocab_len` which iirc should be an empty matrix in the case of floret.

Anyways, I agree that the API would be cleaner if there's more fine-grained typing...although that might also end up too verbose

Reply:
Yeah, maybe not returning anything in queries is an acceptable outcome. I'll wait with merging this PR then and change this, but we need another `impl` in finalfusion-rust to make it work.

Reply:
https://github.com/finalfusion/finalfusion-rust/pull/188

Reply:
Fixed!